flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/resources/KubernetesSharedInformerITCase.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.kubeclient.resources;
+
+import org.apache.flink.kubernetes.KubernetesResource;
+import org.apache.flink.kubernetes.kubeclient.FlinkKubeClient;
+import org.apache.flink.kubernetes.kubeclient.KubernetesConfigMapSharedWatcher;
+import org.apache.flink.kubernetes.kubeclient.KubernetesSharedWatcher.Watch;
+import org.apache.flink.util.TestLogger;
+
+import org.apache.flink.shaded.guava18.com.google.common.collect.ImmutableMap;
+
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+/** IT Tests for the {@link KubernetesSharedInformer}. */
+public class KubernetesSharedInformerITCase extends TestLogger {
+
+    @ClassRule public static KubernetesResource kubernetesResource = new KubernetesResource();
+
+    @Test
+    public void testWatch() throws Exception {
+        final ImmutableMap<String, String> labels =
+                ImmutableMap.of("app", "shared-informer-test-cluster");
+        final String configMapPrefix = "shared-informer-test-cluster";
+        try (FlinkKubeClient client = kubernetesResource.getFlinkKubeClient();
+                KubernetesConfigMapSharedWatcher configMapSharedWatcher =
+                        client.createConfigMapSharedWatcher(labels)) {
+
+            configMapSharedWatcher.run();
+            for (int i = 0; i < 5; i++) {
```

Comment:
The create ConfigMap logics could be deduplicated by introducing an internal method.